ballista/executor/src/flight_service.rs

```diff
@@ -248,3 +289,162 @@ fn from_arrow_err(e: &ArrowError) -> Status {
+                    .get(&path)
+                    .await
+                    .unwrap()
+                    .bytes()
+                    .await
+                    .unwrap();
+                let reader = FileReader::try_new(std::io::Cursor::new(bytes), None)
+                    .map_err(|e| from_arrow_err(&e))?;
+                let schema = reader.schema();
+
+                let (tx, rx) = channel(2);
+
+                let executor_id = self.executor_id.clone();
+                task::spawn_blocking(move || {
+                    if let Err(e) = read_partition(reader, tx) {
+                        warn!(executor_id, error = %e, "error streaming shuffle partition");
+                    }
+                });
+
+                let flight_data_stream = FlightDataEncoderBuilder::new()
+                    .with_max_flight_data_size(MAX_MESSAGE_SIZE)
+                    .with_schema(schema)
+                    .build(ReceiverStream::new(rx))
+                    .map_err(|err| Status::from_error(Box::new(err)));
+
+                Ok(Response::new(
+                    Box::pin(flight_data_stream) as Self::DoGetStream
+                ))
+            }
+        }
+    }
+
+    async fn get_schema(
+        &self,
+        _request: Request<FlightDescriptor>,
+    ) -> Result<Response<SchemaResult>, Status> {
+        Err(Status::unimplemented("get_schema"))
+    }
+
+    async fn get_flight_info(
+        &self,
+        _request: Request<FlightDescriptor>,
+    ) -> Result<Response<FlightInfo>, Status> {
+        Err(Status::unimplemented("get_flight_info"))
+    }
+
+    async fn handshake(
+        &self,
+        _request: Request<Streaming<HandshakeRequest>>,
+    ) -> Result<Response<Self::HandshakeStream>, Status> {
+        let token = uuid::Uuid::new_v4();
+        info!("do_handshake token={}", token);
+
+        let result = HandshakeResponse {
+            protocol_version: 0,
+            payload: token.as_bytes().to_vec().into(),
+        };
+        let result = Ok(result);
+        let output = futures::stream::iter(vec![result]);
+        let str = format!("Bearer {token}");
+        let mut resp: Response<
+            Pin<Box<dyn Stream<Item = Result<_, Status>> + Send + 'static>>,
+        > = Response::new(Box::pin(output));
+        let md = MetadataValue::try_from(str)
+            .map_err(|_| Status::invalid_argument("authorization not parsable"))?;
+        resp.metadata_mut().insert("authorization", md);
+
+        Ok(resp)
+    }
+
+    async fn list_flights(
+        &self,
+        _request: Request<Criteria>,
+    ) -> Result<Response<Self::ListFlightsStream>, Status> {
+        Err(Status::unimplemented("list_flights"))
+    }
+
+    async fn do_put(
+        &self,
+        request: Request<Streaming<FlightData>>,
+    ) -> Result<Response<Self::DoPutStream>, Status> {
+        let mut request = request.into_inner();
+
+        while let Some(data) = request.next().await {
```

Comment:
What does this do?

Reply:
copied from regular implementation, it consumes it 🙂

Reply:
removed